providers/placeholderapi/src/main/java/de/minebench/tresor/providers/placeholderapi/PlaceholderAPIPlaceholders.java

```diff
@@ -0,0 +1,86 @@
+package de.minebench.tresor.providers.placeholderapi;
+
+import de.minebench.tresor.Provider;
+import de.minebench.tresor.services.placeholder.EmptyPlaceholder;
+import de.minebench.tresor.services.placeholder.Placeholder;
+import de.minebench.tresor.services.placeholder.Placeholders;
+import me.clip.placeholderapi.PlaceholderAPI;
+import me.clip.placeholderapi.PlaceholderAPIPlugin;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PlaceholderAPIPlaceholders extends Provider<Placeholders, PlaceholderAPIPlugin> implements Placeholders {
+
+    private PlaceholderAPIPlugin hooked;
+
+    private final Map<String, Placeholder> placeholders = new HashMap<>();
+
+    public PlaceholderAPIPlaceholders() {
+        super(Placeholders.class);
+    }
+
+    @Override
+    public PlaceholderAPIPlugin getHooked() {
+        if (hooked == null)
+            hooked = (PlaceholderAPIPlugin) Bukkit.getPluginManager().getPlugin(getName());
+
+        return hooked;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return getHooked() != null && getHooked().isEnabled();
+    }
+
+    @Override
+    public String getName() {
+        return "PlaceholderAPI";
+    }
+
+    @Override
+    public Map<String, Placeholder> getRegisteredPlaceholders() {
+        return this.placeholders;
+    }
+
+    @Override
+    public Placeholder getPlaceholder(final String key) {
+        final Placeholder placeholder = this.placeholders.get(key);
+
+        if (placeholder == null)
+            return new EmptyPlaceholder();
```

Comment:
Why does this need to return a special `EmptyPlaceholder` class? Wouldn't it make more sense to just return `null` or an `Optional` so it can be easily checked by the querying plugin? I don't really see the benefit of haling that a separate, non-standard class.